internal/armada/scheduling/clusters.go

```diff
@@ -42,6 +65,19 @@ func FilterActiveClusterSchedulingInfoReports(reports map[string]*api.ClusterSch
 	return result
 }
 
+func GroupSchedulingInfoByPool(reports map[string]*api.ClusterSchedulingInfoReport) map[string]map[string]*api.ClusterSchedulingInfoReport {
+	result := map[string]map[string]*api.ClusterSchedulingInfoReport{}
+	for id, report := range reports {
+		poolReports := result[report.Pool]
```

Comment:
If I am reading this loop body correctly, could it be written as:
```
  poolReports, ok := result[report.Pool]
  if !ok { 
    poolReports = map[string]*api.ClusterSchedulingInfoReport{}
    result[report.Pool] = poolReports
  }
  poolReports[id] + report
```

Not suggesting that it gets rewritten that way, primarily checking my understanding of the code.

Reply:
yes, your version is equivalent and more idiomatic, will update it.